static/panes/compiler.js

```diff
@@ -128,21 +128,8 @@ function Compiler(hub, container, state) {
 
     this.initButtons(state);
 
-    var monacoDisassembly = 'asm';
-    // Bandaid fix to not have to include monacoDisassembly everywhere in languages.js
-    if (languages[this.currentLangId]) {
-        switch (languages[this.currentLangId].id) {
-            case 'cuda':
-                monacoDisassembly = 'ptx';
-                break;
-            case 'ruby':
-                monacoDisassembly = 'asmruby';
-                break;
-            case 'mlir':
-                monacoDisassembly = 'mlir';
-                break;
-        }
-    }
+    var monacoDisassembly =
+        (languages[this.currentLangId] ? languages[this.currentLangId].monacoDisassembly : null) || 'asm';
```

Comment:
todo: this should actually be a runtime thing and depend on the compilationresult. But this is the next best thing, so thumbs up from me

Reply:
Right, thanks, yeah :)